File: python/Kal/str1/permutation.py
str1 = "abc"
str2 = "abc"

def permutation(str1, str2):
    dictionary = dict()
    if len(str1) != len(str2):
        return False
    for i in range(len(str1)):
        if str1[i] in dictionary:
            dictionary[str1[i]] += 1
        else:
            dictionary[str1[i]] = 1
    for i in range(len(str2)):
        if str2[i] in dictionary:
            dictionary[str2[i]] -= 1
        else:
            return False
    for key in dictionary:
        if dictionary[key] != 0:
            return False
    return True

answer = permutation(str1, str2)
print(answer)

File: KalPractice/strings/permutation.js
str1= "abcdefg"
str2= "gfedabc"

function permutation(str1, str2) {
    if(str1.length !== str2.length) {
        return console.log(false)
    }
    let hash = {};
    for(let i = 0; i < str1.length; i++) {
        if(hash[str1[i]]) {
            hash[str1[i]] += 1;
        } else {
            hash[str1[i]] = 1;
        }
    }
    for(let i = 0; i < str2.length; i++) {
        if(hash[str2[i]]) {
            hash[str2[i]] -=1;
        } else {
            return console.log(false)
        }
    }
    return console.log(true)
}

permutation(str1, str2);